apps/web/src/app/wallet/hooks/use-init-transfer.ts

```diff
@@ -99,26 +102,34 @@ export const useInitTransfer = ({ params, enabled }: InitTransferProps) => {
 
     if (!params.type) return
 
-    modalService.open({
-      key: loadingTransferParamsModalKey,
-      variantOptions: {
-        variant: 'loading',
-        isLocked: true,
-      },
-    })
-
     if (isTransferTypeParams(params)) {
+      modalService.open({
+        key: loadingTransferParamsModalKey,
+        variantOptions: {
+          variant: 'loading',
+          isLocked: true,
+        },
+      })
       await getTransferOptions.mutateAsync(params)
-    } else if (isNftTypeParams(params)) {
-      await getTransferOptions.mutateAsync(params)
+    } else if (isNftClaimTypeParams(params)) {
+      try {
+        const result = await walletService.getNftClaimOptions(params.session_id, params.resource)
```

Comment:
```suggestion
        const result = await getNftClaimOptions.mutateAsync(params.session_id, params.resource)
```

I would recommend wrapping this getNftClaimOptions into a react query mutation you could also set on this file like:

```
const getNftClaimOptions = useGetNftClaimOptions({
      onSuccess: (result) => {
         const nftData = result.data.nft
         handleClaimNft(nftData, params.session_id, params.resource)
      } 
      onError: (error)=> {
         ErrorHandling.handleError({ error })
         exit({ closeModal: true })
      }
})
```

This way, the `handleTransferParams` will not need to have `exit` and `toast` as deps